File: nlp/text.py
import time
import collections

filename = "corpus.txt"
string = ''
word = ''
wordMap = dict()


def addToMap(word):
    if word not in wordMap:
        wordMap[word] = 1
    else:
        wordMap[word] += 1


def cleanWord(word):
    word = word.replace(".", "")
    word = word.replace(",", "")
    word = word.replace("\"", "")
    word = word.replace("!", "")
    word = word.replace("?", "")
    word = word.replace(";", "")
    return word


with open(filename) as f:
    string = f.read()

start = start_time = time.time()

for word in string.lower().split():
    addToMap(cleanWord(word))

# alt 2
# for char in string:
#     if char == ' ':
#         if string == '':
#             continue
#         else:
#             addToMap(cleanWord(word))
#             word = ''
#     elif char in ['.', ',', '\"', '?', '!', ':', ';']:
#         continue
#     else:
#         word += char.lower()

finish = time.time()
elapsed = finish - start
print("Elapsed time:", elapsed, "seconds")

d = collections.Counter(wordMap)
for word, count in d.most_common(10):
    print(word, ": ", count)

File: nlp/Text.java
import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Path;
import java.util.HashMap;
import java.util.Map;

public class Text {
    private static String filename = "corpus.txt";
    private static String fileContent = "";
    private static String word = "";
    private static Map<String, Integer> wordMap = new HashMap<>();

    private static void addToMap(String word) {
        wordMap.put(word, wordMap.getOrDefault(word, 0) + 1);
    }

    private static String cleanWord(String word) {
        word = word.replace(".", "");
        word = word.replace(",", "");
        word = word.replace("\"", "");
        word = word.replace("!", "");
        word = word.replace("?", "");
        word = word.replace(";", "");
        return word;
    }

    public static void main(String[] args) {
        long start = System.currentTimeMillis();

        try {
            fileContent = Files.readString(Path.of("./" + filename));
        } catch (IOException e) {
            e.printStackTrace();
        }

        for (String word : fileContent.toLowerCase().split(" ")) {
            addToMap(cleanWord(word));
        }

        long finish = System.currentTimeMillis();
        long elapsed = finish - start;
        System.out.println("Elapsed time: " + elapsed + " milliseconds");

        start = System.currentTimeMillis();

        for (char ch : fileContent.toCharArray()) {
            if (ch == ' ') {
                if (word.equals("")) {
                    continue;
                } else {
                    addToMap(cleanWord(word));
                    word = "";
                }
            } else if (".,\"?!:;".contains(String.valueOf(ch))) {
                continue;
            } else {
                word += Character.toLowerCase(ch);
            }
        }

        finish = System.currentTimeMillis();
        elapsed = finish - start;
        System.out.println("Elapsed time: " + elapsed + " milliseconds");

        // implement sorting of hashmap
        // for (int i = 0; i < 10; i++) {
        //     System.out.println(entry.getKey() + ": " + entry.getValue());
        // }
    }
}

